splunk_add_on_ucc_framework/ucc_ui_lib/src/main/webapp/components/EntityModal.jsx

```diff
@@ -2,34 +2,58 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Button from '@splunk/react-ui/Button';
 import Modal from '@splunk/react-ui/Modal';
+import styled from 'styled-components';
+import WaitSpinner from '@splunk/react-ui/WaitSpinner';
+
+import { MODE_CLONE, MODE_CREATE, MODE_EDIT } from '../constants/modes';
 import BaseFormView from './BaseFormView';
 
+const ModalWrapper = styled(Modal)`
+    width: 800px
+`;
+
 class EntityModal extends Component {
     constructor(props) {
         super(props);
         this.form = React.createRef();
+        this.state = {isSubmititng:false};
+        
+        if(props.mode === MODE_CREATE){
+            this.buttonText ="Add";
+        }
+        else if(props.mode === MODE_CLONE){
+            this.buttonText = "Save";
+        }
+        else if(props.mode ===MODE_EDIT){
+            this.buttonText = "Update";
+        }
+        else{
+            this.buttonText = "Submit";
+        }
     }
```

Comment:
Can we use switch here instead of if..else if..

Reply:
We can keep _else if_ as there are only _2 else if_ conditions.